runtime/doc/starting.txt

```diff
@@ -349,11 +347,21 @@ argument.
 							*--api-info*
 --api-info 	Print msgpack-encoded |api-metadata| and exit.
 
+							*--embed*
+--embed		Use stdin/stdout as a msgpack-RPC channel.  This makes it easy
+		for applications to embed and control Nvim via the |rpc-api|.
+                Implies |--headless|.  Equivalent to: >
+			nvim --headless --cmd "call stdioopen({'rpc': v:true})"
+<
+                See also |channel-stdio|.
+
 							*--headless*
---headless 	Do not start the built-in UI.
-                See |channel-stdio| for how to use stdio for other purposes
-                instead.
-		See also |silent-mode|, which does start a (limited) UI.
+--headless 	Do not start the built-in UI.  Allows stdio to be used as
+                a communication channel. |channel-stdio| To detect this state
```

Comment:
My old comment was marked as outdated but is still not resolved: `nvim_list_uis()` does not reliably detect `--headless` ("this state"). It's better to say what it actually detects: "To detect in a script whether an UI is available"